libraries/Microsoft.Bot.Builder.Azure.Queues/QueuesStorage.cs

```diff
@@ -0,0 +1,81 @@
+using Azure.Storage.Queues.Models;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Azure.Queues
+{
+    /// <summary>
+    /// Service used to add messages to an Azure.Storage.Queues.
+    /// </summary>
+    public class QueuesStorage
+    {
+        private JsonSerializerSettings _jsonSettings;
+        private bool _createQueuIfNotExists = true;
+        private readonly QueueClient _queueClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QueuesStorage"/> class.
+        /// </summary>
+        /// <param name="queuesStorageConnectionString">Azure Storage connection string.</param>
+        /// <param name="queueName">Name of the storage queue where entities will be queued.</param>
+        /// <param name="jsonSerializerSettings">If passing in custom JsonSerializerSettings, we 
+        /// recommend the following settings:
+        /// <para>jsonSerializer.TypeNameHandling = TypeNameHandling.None.</para>
+        /// <para>jsonSerializer.NullValueHandling = NullValueHandling.Ignore.</para>
+        /// </param>
+        public QueuesStorage(string queuesStorageConnectionString, string queueName, JsonSerializerSettings jsonSerializerSettings = null)
+        {
+            if (string.IsNullOrEmpty(queuesStorageConnectionString))
+            {
+                throw new ArgumentNullException(nameof(queuesStorageConnectionString));
+            }
+
+            if (string.IsNullOrEmpty(queueName))
+            {
+                throw new ArgumentNullException(nameof(queueName));
+            }
+
+            _jsonSettings = jsonSerializerSettings ?? new JsonSerializerSettings
+                {
+                    TypeNameHandling = TypeNameHandling.None,
+                    NullValueHandling = NullValueHandling.Ignore
+                };
+
+            _queueClient = new QueueClient(queuesStorageConnectionString, queueName);
+        }
+
+        /// <summary>
+        /// Queue and Activity, with option in the Activity.Value to Azure.Storage.Queues.
+        /// </summary>
+        /// <param name="activity">This is expected to be an <see cref="Activity"/> retrieved from a call to 
+        /// activity.GetConversationReference().GetContinuationActivity().  This enables restarting the conversaton
+        /// using BotAdapter.ContinueConversationAsync.</param>
+        /// <param name="visibilityTimeout">Default value of 0.  Cannot be larger than 7 days.</param>
+        /// <param name="timeToLive">Specifies the time-to-live interval for the message.</param>
+        /// <param name="cancellationToken">Cancellation token for the async operation.</param>
+        /// <returns><see cref="SendReceipt"/> from the QueueClient SendMessageAsync operation.</returns>
+        public async Task<SendReceipt> QueueActivityAsync(Activity activity, TimeSpan? visibilityTimeout = null, TimeSpan? timeToLive = null,  CancellationToken cancellationToken = default)
```

Comment:
Remove SendReceipt - or turn it into a string. Let's not leak the dependency.

Reply:
done